### PR TITLE
fix(node/path): Remove toFileUrl

### LINKED ISSUE #18177: `node:path` module contains non-node symbols
Namely exports from https://deno.land/std@0.179.0/path/glob.ts, https://deno.land/std@0.179.0/path/separator.ts and some others. Also missing circular references like `path.posix.posix === path.posix`.
```ts
import path from "node:path";
console.assert(path.SEP === undefined);
console.assert(path.SEP_PATTERN === undefined);
console.assert(path.fromFileUrl === undefined);
console.assert(path.globToRegExp === undefined);
console.assert(path.isGlob === undefined);
console.assert(path.joinGlobs === undefined);
console.assert(path.normalizeGlob === undefined);
console.assert(path.toFileUrl === undefined);
console.assert(path.posix.fromFileUrl === undefined);
console.assert(path.posix.toFileUrl === undefined);
console.assert(path.win32.fromFileUrl === undefined);
console.assert(path.win32.toFileUrl === undefined);
console.assert(path.posix.posix === path.posix);
console.assert(path.posix.win32 === path.win32);
console.assert(path.win32.posix === path.posix);
console.assert(path.win32.win32 === path.win32);
if (Deno.build.os == "windows") {
  console.assert(path === path.win32);
} else {
  console.assert(path === path.posix);
}
```

### PR DESCRIPTION
**Pull request by RyanClementsHax**
- Remove the toFileUrl function from node/path since it doesn't exist there
- Add more assertions against referential invariants of the path export of node
- Fixes #18177

### CODE REVIEW

File: ext/node/polyfills/worker_threads.ts [hunk 1]
```diff
@@ -32,6 +32,67 @@ export interface WorkerOptions {
   workerData?: unknown;
 }
 
+const WHITESPACE_ENCODINGS: Record<string, string> = {
```
Comment (RyanClementsHax) [inline ext/node/polyfills/worker_threads.ts:35]: I'm not sure where to put all of this `toFileUrl` code. I considered putting it in https://github.com/denoland/deno/blob/main/ext/node/polyfills/_utils.ts but the dependency of this code on `node/path`'s `isAbsolute` function complicates the file dependencies as then everything that includes that file now transitively depends on `node/path`.

For now I kept it in this file, but would like other input on where this logic might best live.

Comment (kt3k) [reply ext/node/polyfills/worker_threads.ts:35]: Putting these here looks fine to me

**Review by kt3k — APPROVED** [review]
LGTM. Thanks!

**Comment by kt3k** [comment]
Note: I'll do some clean up work on `cli/tests/node_compat/test/parallel/test-path.js` later (as that file needs to be in sync with https://github.com/nodejs/node/blob/main/test/parallel/test-path.js ).

**Pull request merged**